### ISSUE TITLE
Include filter for filtering default innerblocks template

### ISSUE THREAD

**Issue by Lovor01** [WordPress/gutenberg#30936]
## What problem does this address?
Currently, I see no other way than to modify block edit function through `blocks.registerBlockType` or `editor.BlockEdit` filters in order to modify default template for a content which is inserted in block's InnerBlocks component. Since this action requires copying a lot of code from original block, I find it awkward to use.

## What is your proposed solution?
Include **filter** which filters default InnerBlock's template. It would be even better if developer could change this filter dynamically, enabling him to adjust content which is being inserted in block's InnerBlocks component depending on block's context, user interaction and similarly

**Comment by gziolo** [comment]
Related comment from @youknowriad on a PR https://github.com/WordPress/gutenberg/pull/30749#issuecomment-818670858 that is applicable here as well:

> In a previous PR we added a "templateLock" attribute to the "column" block to be able to edit it there. In a lot of ways "templateLock" is similar to "allowedBlocks" and "template". An attribute allows templates to override these things for specific instances which is arguably more powerful.
> 
> I'm personally not at ease with using filters here (or any js filter for the record), it breaks too often depending on how/when scripts are loaded, components rerendered...

We need a more general way to control what is being passed to the `InnerBlock` component or with `useInnerBlockProps` hook.

**Comment by Lovor01** [comment]
I agree with you. How can I achieve this currently (i.e. changing what template will be inserted when user clicks on inserter plus button in `InnnerBlocks` , depending on block parent, user choice, state of the editor etc.)?
